hadoop-hdds/framework/src/main/java/org/apache/hadoop/hdds/utils/DBCheckpointServlet.java

```diff
@@ -92,7 +97,8 @@ private boolean hasPermission(UserGroupInformation user) {
     if (aclEnabled && isSpnegoEnabled) {
       return allowedUsers.contains(OZONE_ADMINISTRATORS_WILDCARD)
           || allowedUsers.contains(user.getShortUserName())
-          || allowedUsers.contains(user.getUserName());
+          || allowedUsers.contains(user.getUserName())
+          || !Sets.intersection(new HashSet<>(allowedGroups), new HashSet<>(user.getGroups())).isEmpty();
```

Comment:
Nit: I kinda wish the `Sets.intersection(new HashSet<>(allowedGroups), new HashSet<>(user.getGroups())).isEmpty()` could be extracted to a small helper method like `isUserInAdminGroup` to decrease cognitive load when looking at the whole boolean expression. It's up to your preference.

Reply:
Thank you, 
I decided to make one step further and extracted the duplicated code that checks admin users to a separate class